docs/concepts/internationalization.md

```diff
@@ -0,0 +1,75 @@
+Internationalization and localization
+=====================================
+
+Internationalization, i18n for short, is a way of designing code so that it requires no changes to be adapted for various languages. Localization is the process of adapting displayed text to the user's language.
+
+I18n is implemented using _translation keys_. A translation key is a string that identifies a piece of displayable text in no specific language. For example, `tile.dirt.name` is the translation key referring to the name of the Dirt block. This way, displayable text may be referenced with no concern for a specific language. The code requires no changes to be adapted in a new language.
+
+Localization will happen in the game's locale. In a Minecraft client the locale is specified by the language settings. On a dedicated server, the only supported locale is en_US. A list of available locales can be found on the [Minecraft Wiki](https://minecraft.gamepedia.com/Language#Available_languages).
+
+Language files
+--------------
+
+Language files are located by `assets/[namespace]/lang/[locale].lang` (e.g. the US English translation for `examplemod` would be `assets/examplemod/lang/en_us.lang`). Resource pack format 3 requires the locale name to be lowercased. The file format is simply lines of key-value pairs separated by `=`. Lines starting with `#` are ignored. Lines without a separator are ignored. The file must be encoded in UTF-8.
+
+```properties
+# items
+item.examplemod.example_item.name=Example Item Name
+
+# blocks
+tile.examplemod.example_block.name=Example Block Name
+
+# commands
+commands.examplemod.examplecommand.usage=/example <value>
+```
+
+Including the comment `#PARSE_ESCAPES` anywhere in the file will enable the slightly different and more complex Java properties file format. This format is required to support multiline values.
+
+!!! note
+    Enabling `#PARSE_ESCAPES` changes various aspects of the parsing. Among other things, the `:` character will be treated as a key-value separator. It must either be excluded from translation keys or escaped using `\:`.
+
+Usage with Blocks and Items
```

Comment:
Should this section include a brief note about the game looking for a key with `.name` appended by default?